rend3-anim/src/lib.rs

```diff
@@ -0,0 +1,274 @@
+            let joint_nodes_topological_order: Vec<NodeIndex> = instance
+                .topological_order
+                .iter()
+                .map(|node_idx| NodeIndex(*node_idx))
+                .filter(|node_idx| skin_nodes.contains(node_idx))
+                .collect();
+
+            let skeletons: Vec<SkeletonHandle> = instance
+                .nodes
+                .iter()
+                .flat_map(|node| &node.inner.object)
+                .flat_map(|object| &object.inner.armature)
+                .filter(|armature| armature.skin_index == skin_index.0)
+                .flat_map(|armature| &armature.skeletons)
+                .cloned()
+                .collect();
+
+            skin_data.insert(
+                skin_index,
+                PerSkinData {
+                    node_to_joint_idx,
+                    joint_nodes_topological_order,
+                    skeletons,
+                },
+            );
+        }
+
+        AnimationData {
+            skin_data,
+            animation_skin_usage,
+        }
+    }
+}
+
+/// Helper trait that exposes a generic `lerp` function for various `glam` types
+pub trait Lerp {
+    fn lerp(self, other: Self, t: f32) -> Self;
+}
+impl Lerp for Vec3 {
+    fn lerp(self, other: Self, t: f32) -> Self {
+        self.lerp(other, t)
+    }
+}
+impl Lerp for Quat {
+    fn lerp(self, other: Self, t: f32) -> Self {
+        // Uses Normalized Linear Interpolation (a.k.a. nlerp) as slerp replacement
+        // See: *"Understanding Slerp, Then Not Using It"*
+        // http://number-none.com/product/Understanding%20Slerp,%20Then%20Not%20Using%20It/
+        self.lerp(other, t).normalize()
+    }
+}
+
+/// Samples the data value for an animation channel at a given time. Will
+/// interpolate between the two closest keyframes.
+fn sample_at_time<T: Lerp + Copy>(channel: &AnimationChannel<T>, current_time: f32) -> T {
+    let next_idx = channel
+        .times
+        .iter()
+        .position(|time| *time > current_time)
+        .unwrap_or(channel.times.len() - 1);
+    let prev_idx = next_idx.saturating_sub(1);
+
+    let interp_factor = f32::clamp(
+        (current_time - channel.times[prev_idx]) / (channel.times[next_idx] - channel.times[prev_idx]),
+        0.0,
+        1.0,
+    );
+
+    channel.values[prev_idx].lerp(channel.values[next_idx], interp_factor)
+}
+
+/// Sets the pose of the meshes at the given scene by using the animation at
+/// index `animation_index` at a given `time`. The provided time gets clamped to
+/// the valid range of times for the selected animation.
+pub fn pose_animation_frame(
+    renderer: &Renderer,
+    scene: &LoadedGltfScene,
+    instance: &GltfSceneInstance,
+    animation_data: &AnimationData,
+    animation_index: usize,
+    time: f32,
+) {
+    let animation = &scene.animations[animation_index];
+    let time = time.clamp(0.0, animation.inner.duration);
+
+    for (skin_index, per_skin_data) in animation_data.skin_data.iter() {
```

Comment:
Prefer iterating over a reference to a container vs a raw call to iter 
```suggestion
    for (skin_index, per_skin_data) in &animation_data.skin_data {
```